### DISCUSSION TITLE
Building Editor Widget Similar to Diff Editor

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11783] [Q&A]
Hi,
I'm trying to build an EditWidget that is similar to DiffEditor. Do we have any reference in Theia similar to this ?
I was trying to find the source code for DiffEditor but seemed to point to a different package that is not in theia (Correct me if I'm wrong).
It would be great if you point in the right direction.

![image](https://user-images.githubusercontent.com/80001447/196360588-0c7468f3-e8d7-40b6-b3bb-004404cb5b40.png)

Also is it possible to open two editor side by side by default similar to this using a command. Also do we have any features similar to this in theia that I could refer to.
![image](https://user-images.githubusercontent.com/80001447/196362298-8665c854-69a8-463c-87d9-ca48a8c2ce9b.png)

Thanks

**Comment by msujew** [comment]
Hey @maheshrajrp,

the diff editor isn't really built by us, we just build our infrastructure on top. The actual functionality is provided by Microsoft's Monaco editor, which they use in vscode.

For your second question, you can take a look at the [`WidgetOptions`](https://github.com/eclipse-theia/theia/blob/7f51450b88967fe2f96f4f895357413bf311ad16/packages/core/src/browser/shell/application-shell.ts#L2032-L2049) which get passed into the [`ApplicationShell#addWidget`](https://github.com/eclipse-theia/theia/blob/7f51450b88967fe2f96f4f895357413bf311ad16/packages/core/src/browser/shell/application-shell.ts#L805-L837) method. It allows you to open a widget to the left/right of another widget.
